Source/WebKit/WebProcess/Storage/WebSWClientConnection.cpp

```diff
@@ -172,8 +172,24 @@ void WebSWClientConnection::matchRegistration(SecurityOriginData&& topOrigin, co
         return;
     }
 
-    runOrDelayTaskForImport([this, callback = WTFMove(callback), topOrigin = WTFMove(topOrigin), clientURL]() mutable {
-        sendWithAsyncReply(Messages::WebSWServerConnection::MatchRegistration { topOrigin, clientURL }, WTFMove(callback));
+    struct CallbackCaller {
+        CallbackCaller(RegistrationCallback&& callback)
+            : callback(WTFMove(callback)) { }
+        CallbackCaller(CallbackCaller&&) = default;
+        CallbackCaller& operator=(CallbackCaller&&) = default;
+        ~CallbackCaller()
+        {
+            // Fail to register if this WebSWClientConnection is destroyed before runOrDelayTaskForImport's callback is called,
+            // such as if the network process crashes before we send the message to it.
+            if (callback)
+                callback(std::nullopt);
+        }
+        RegistrationCallback callback;
+    };
+
+    CallbackCaller caller(WTFMove(callback));
```

Comment:
Can we use one of the existing CompletionHandler existing class, like CompletionHandlerWithFinalizer?